lib/ansible/modules/cloud/memset/memset_lb_service.py

```diff
@@ -0,0 +1,342 @@
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: memset_lb_service
+author: "Simon Weald (@glitchcrab)"
+version_added: "2.9"
+short_description: Manage Memset loadbalancer services
+notes:
+  - A loadbalancer service is logically the Internet-facing 'frontend'. This must
+    be backed by one or more servers using the C(memset_lb_server) module.
+  - An API key generated via the Memset customer control panel is needed with the
+    following minimum scope - I(loadbalancer.service.add), I(loadbalancer.service.info),
+    I(loadbalancer.service.list), I(loadbalancer.service.remove),
+    I(loadbalancer.service.update), I(server.info).
+description:
+    - Manage Memset loadbalancer services
+options:
+    state:
+        required: false
+        default: present
+        description:
+            - Indicates desired state of resource. Defaults to present.
+            - When deleting a service there must be no servers currently attached to it as this will raise an error.
+        choices: [ absent, present ]
+    api_key:
+        required: true
+        description:
+            - The API key obtained from the Memset control panel.
+    enabled:
+        required: false
+        default: true
+        type: bool
+        description:
+            - Whether the service is enabled or not. Defaults to True.
+    load_balancer:
+        required: true
+        description:
+            - The name of the load balancer - this is the product name e.g. C(lbtestyaa1).
+    port:
+        required: true
+        description:
+            - The port to be exposed to the Internet.
+            - Must be in the range 1 > 65535 (inclusive).
+    protocol:
+        required: true
+        description:
+            - The protocol to be used by the load balacer.
+        choices: [ tcp, http, https ]
+    service_name:
+        required: true
+        description:
+            - Unique name to identify the service by (must be unique). Changing this will cause a new service to be created.
+            - It can only consist of letters, numbers, underscores and hyphens and must be a maximum of 64 characters.
+        aliases: [ 'name' ]
+    virtual_ip:
+        required: false
+        description:
+            - The IP address to expose the service on (must be assigned to the loadbalancer product).
+            - If not provided, it will default to the primary IP of the loadbalancer.
+'''
+
+EXAMPLES = '''
+- name: create a loadbalanced service
+  memset_lb_service:
+    state: present
+    api_key: 5eb86c9196ab03919abcf03857163741
+    load_balancer: lbtestyaa1
+    port: 443
+    protocol: https
+    service_name: my_https_service
+    virtual_ip: 1.2.3.4
+  delegate_to: localhost
+'''
+
+RETURN = '''
+---
+memset_api:
+  description: Info from the Memset API
+  returned: when changed or state == present
```

Comment:
Not blocker but it seems to me that state=absent does not return the "memset_api" it had just deleted. But return the resource which has been deleted would be useful for integration tests. Also see this https://github.com/ansible/ansible/blob/devel/test/integration/targets/vultr_user/tasks/main.yml#L182-L225 as an example.